portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -42,12 +42,21 @@ public void doGet(HttpServletRequest request, HttpServletResponse response) thro
 
   @Override
   public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
-    String name = request.getParameter("name");
-    String body = request.getParameter("body");
-    if (name == null || body == null) { // Someone sending a bad form.
-      return;
+    final String name = request.getParameter("name");
+    final String body = request.getParameter("body");
+    if (!isSaneForm(name, body)) {
```

Comment:
Consider adding an error response code and message. 4XX errors are for client errors.

https://developer.mozilla.org/en-US/docs/Web/HTTP/Status